tensornetwork/tn_keras/entangler.py

```diff
@@ -102,24 +102,32 @@ def is_perfect_root(n, n_nodes):
 
     super(DenseEntangler, self).build(input_shape)
 
-    # Ensure input dim and output dim match
-    assert (
-        input_shape[-1] == self.output_dim
-    ), f'Input dim {input_shape[-1]} not equal to output dim {self.output_dim}'
-
     # Ensure the Entangler dimensions will work
     assert (
         is_perfect_root(input_shape[-1], self.num_legs)
     ), f'Input dim {input_shape[-1]}**(1. / {self.num_legs}) must be round.'
 
+    assert (
+        is_perfect_root(self.output_dim, self.num_legs)
+    ), f'Output dim {self.output_dim}**(1. / {self.num_legs}) must be round.'
+
     self.leg_dim = round(input_shape[-1]**(1. / self.num_legs))
-    self.num_nodes = self.num_levels
+    self.out_leg_dim = round(self.output_dim**(1. / self.num_legs))
+    self.num_nodes = self.num_levels * (self.num_legs - 1)
 
     for i in range(self.num_nodes):
+      current_level = i // (self.num_legs - 1)
+      node_out_dim = self.out_leg_dim
+      node_connect_dim = self.out_leg_dim
```

Comment:
I think we should either (a) set this to the min of `out_leg_dim` and `in_leg_dim`, so we use the smaller of the two options or (b) expose a separate `bond_dim` parameter the user can tune and supply some guidance around reasonable values. (a) seems more clear for end users, while (b) seems better in a research context (where we are likely to want to try different `bond_dim` values and check performance.